kaldb/src/test/java/com/slack/kaldb/metadata/service/ServicePartitionMetadataTest.java

```diff
@@ -79,4 +122,113 @@ public void testValidServicePartitionMetadata() {
         .isThrownBy(
             () -> new ServicePartitionMetadata(start.toEpochMilli(), end.toEpochMilli(), null));
   }
+
+  @Test
+  public void testOneServiceOnePartition() {
+    final String name = "testService";
+    final String owner = "serviceOwner";
+    final long throughputBytes = 1000;
+    final ServicePartitionMetadata partition = new ServicePartitionMetadata(100, 200, List.of("1"));
+
+    ServiceMetadata serviceMetadata =
+        new ServiceMetadata(name, owner, throughputBytes, List.of(partition));
+
+    serviceMetadataStore.createSync(serviceMetadata);
+    await().until(() -> serviceMetadataStore.listSync().size() == 1);
+
+    List<ServicePartitionMetadata> partitionMetadata =
+        findPartitionsToQuery(serviceMetadataStore, 101, 199, name);
```

Comment:
A couple comments for each of these sections explain what you're testing would be helpful.